Circuit/Components/VacuumTubes/Pentode.cs

```diff
@@ -0,0 +1,119 @@
+﻿using ComputerAlgebra;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace Circuit.Components
+{
+    [Category("Vacuum Tubes")]
+    [DisplayName("Pentode")]
+    public class Pentode : Component
+    {
+        private Terminal _plate, _grid, _grid2, _cathode;
+
+        private double _mu = 10.7;
+        [Serialize, Category("Koren"), Description("Voltage gain.")]
+        public double Mu { get { return _mu; } set { _mu = value; NotifyChanged(nameof(Mu)); } }
+
+        private double _kg1 = 1672.0;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kg1 { get { return _kg1; } set { _kg1 = value; NotifyChanged(nameof(Kg1)); } }
+
+        private double _kg2 = 4500;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kg2 { get { return _kg2; } set { _kg2 = value; NotifyChanged(nameof(Kg2)); } }
+
+        private double _kp = 41.16;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kp { get { return _kp; } set { _kp = value; NotifyChanged(nameof(Kp)); } }
+
+        private double _kvb = 12.7;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kvb { get { return _kvb; } set { _kvb = value; NotifyChanged(nameof(Kvb)); } }
+
+        private double _ex = 1.310;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Ex { get { return _ex; } set { _ex = value; NotifyChanged(nameof(Ex)); } }
+
+        public Pentode()
+        {
+            _plate = new Terminal(this, "P");
+            _grid = new Terminal(this, "G");
+            _grid2 = new Terminal(this, "G2");
+            _cathode = new Terminal(this, "K");
+        }
+
+        public override IEnumerable<Terminal> Terminals
+        {
+            get
+            {
+                yield return _plate;
+                yield return _grid;
+                yield return _grid2;
+                yield return _cathode;
+            }
+        }
+
+        protected internal override void LayoutSymbol(SymbolLayout Sym)
+        {
+            Sym.AddTerminal(_plate, new Coord(0, 25), new Coord(0, 10));
+            Sym.AddWire(new Coord(-10, 10), new Coord(10, 10));
+
+            Sym.AddTerminal(_grid, new Coord(-20, -5), new Coord(-12, -5));
+            Sym.AddTerminal(_grid2, new Coord(20, 0), new Coord(12, 0));
+            for (int i = -10; i < 10; i += 8)
+            {
+                Sym.AddWire(new Coord(i, -5), new Coord(i + 4, -5));
+                Sym.AddWire(new Coord(i, 0), new Coord(i + 4, 0));
+                Sym.AddWire(new Coord(i, 5), new Coord(i + 4, 5));
+
+            }
+            Sym.AddTerminal(_cathode, new Coord(-10, -25), new Coord(-10, -12), new Coord(-8, -10), new Coord(8, -10), new Coord(10, -12));
+
+            Sym.DrawArc(EdgeType.Black, new Coord(0, 5), 20d, 0, Math.PI, Direction.Counterclockwise);
```

Comment:
Should there be an AddArc, so it can add this part of the symbol to the bounding box? The difference between e.g. `AddCircle` and `DrawCircle` is that the former includes the shape in the bounding box of the symbol, while the latter does not. I think these arcs are an important part of the circle.

Reply:
I know the difference, I was intentionally avoiding it, as it's not needed in this exact case ([YAGNI](https://xkcd.com/974/) 😄), but it looks like i need to add a formula to convert Polar -> Cartesian, because I can't remember if we have it implemented somewhere already.

Reply:
If it really doesn't affect the symbol bounding box (I guess because of the terminals?), you can leave it for now I suppose. But if we unify arcs and circles (which I think is a good thing to do), then we need this anyways...

I think it's fine to implement this by just approximating the circle, no need to get too fancy with solving for the bounding box of an arc :) Just something like:

    for (float i = 0.0f; i <= 1.0f; i += 0.01f) {
        float theta = start + (end - start) * i;
        InBounds(center + (Math.Cos(theta), Math.Sin(theta)) * radius);  // TODO: Not actually correct syntax for a vector add...
    }

Reply:
Actually, it looks like might really simple, we only need to check for startPoint, endPoint, and all points for angles in [ 0, pi/2, pi, 1.5*pi ] that are part of the arc.

Reply:
There is a small caveat - bounds are using Coords, which means we need to implement some rounding after converting form polar to Cartesian 🤔

Reply:
And I found another one - current circle API uses two point method, when you provide two corners of a bounding box and it draws an ellipse inside that box. Merging the two together would require changing that to a center point + radius model or using same, two point approach for arcs, which I deliberately avoided as it can easily get messy (consider large distance between two points and small radius).

Reply:
Let's just leave it for now. The difference you pointed out is big one.